Pcap++/header/PcapFilter.h

```diff
@@ -243,19 +243,26 @@ namespace pcpp
 	class IPFilter : public IFilterWithDirection
 	{
 	private:
-		std::string m_Address;
+		static constexpr int NO_Len_VALUE = 0;
+		static constexpr const char* NO_IPv4Mask_VALUE = "";
```

Comment:
`static` and `constexpr` should not be used together
another way you can write `constexpr std::string_view DEFAULT_IPV4MASK_EMPTY = "";` here.

Reply:
> `static` and `constexpr` should not be used together

Huh? Why? Since they do different things? `static` declaring that the variable is shared between all the instances and `constexpr` that the variable can be computed at compile time. Also removing `static` from the expression leads to a compiler error about `constexpr` being invalid there.

> `constexpr std::string_view DEFAULT_IPV4MASK_EMPTY = "";` here.

`string_view` is part of the C++17 standard, not C++11.

Reply:
Yeah, I forgot we are C++11, however as this string is just a "empty" string, I don't see any benefit to keep it static, the empty string should be already inlined during compilation by constexpr.

Reply:
You can't not make it a member constexpr field and not be static. The compiler throws a fit.

![image](https://github.com/seladb/PcapPlusPlus/assets/20269350/4211022a-71d8-4639-8bf8-c618775cedcf)

Reply:
you could put `constexpr const char* EMTPY_STRING = "";` outside the class. again, no point in using `static` for an empty string here, I would rather you fill `""` in the code directly.

Reply:
> you could put constexpr const char* EMTPY_STRING = ""; outside the class.

Yes, but that throws away the entire point of the change. The primary point was to centralize the definition of the magic value used for "no mask" so you don't have random unspecified magic values sprinkled in the code. Not just to have a `EMPTY_STRING` variable, and I agree that having an `EMPTY_STRING` is pointless and you might as well use `""` where you need an empty string.

A thing that popped up is that apparently `inline` variables are C++17 (and `constexpr` implies inline in C++17 and after), so I will revert the change due to this as pre-C++17 `constexpr` variables in header files have ODR problems.

But that still does not answer my question of what your reasoning is that `static` and `constexpr` should not be used together inside a class scope?

Reply:
>  pre-C++17 `constexpr` variables in header files have ODR problems.

nice catch. seems it may cause an undefined behavior.

> static and constexpr should not be used together inside a class scope?

program-wide, yes it is okay. I didn't say clearly before, but I already explained why I think `static` is unnecessary here.

Reply:
> nice catch. seems it may cause an undefined behavior.

Not undefined actually, just that every translation unit will get its own copy. Doesn't matter much for a constant but it is still extra memory. This SO [answer](https://stackoverflow.com/a/50489130/11922936) explains it in more detail.

Reply:
sure, seems you are correct.